src/main/java/com/softserveinc/dokazovi/controller/AuthorController.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+import static com.softserveinc.dokazovi.controller.EndPoints.AUTHOR_GET_DOCTOR_BY_ID;
+import static com.softserveinc.dokazovi.controller.EndPoints.DOCTOR;
+import static com.softserveinc.dokazovi.controller.EndPoints.DOCTORS;
+
+@RestController
+@RequestMapping(DOCTOR)
+@RequiredArgsConstructor
+public class AuthorController {
+
+	private final AuthorService authorService;
+
+
+	@PostMapping()
+	@PreAuthorize("hasAuthority('CREATE_AUTHOR')")
+	@ApiOperation(value = "Сreate author",
+			authorizations = {@Authorization(value = "Authorization")})
+	@ApiResponses(value = {
+			@ApiResponse(code = 201, message = HttpStatuses.CREATED, response = AuthorDTO.class),
+			@ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST)
+	})
+	public ResponseEntity<AuthorDTO> createAuthor(@Valid @RequestBody AuthorDTO author,
+			@AuthenticationPrincipal UserPrincipal userPrincipal) {
+		return ResponseEntity
+				.status(HttpStatus.CREATED)
+				.body(authorService.save(author, userPrincipal));
+	}
+
+	@PutMapping()
+	@PreAuthorize("hasAuthority('UPDATE_AUTHOR')")
+	@ApiOperation(value = "update author",
+	authorizations = {@Authorization(value = "Authorization")})
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = HttpStatuses.OK, response = AuthorDTO.class),
+			@ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST)
+	})
+	public ResponseEntity<AuthorDTO> updateAuthor(@RequestBody AuthorDTO author,
+			@AuthenticationPrincipal UserPrincipal userPrincipal) {
+		return ResponseEntity
+				.status(200)
+				.body(authorService.update(author, userPrincipal));
+	}
+
+	@DeleteMapping(AUTHOR_GET_DOCTOR_BY_ID)
+	@PreAuthorize("hasAuthority('DELETE_AUTHOR')")
+	@ApiOperation(value = "remove author",
+			authorizations = {@Authorization(value = "Authorization")})
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = HttpStatuses.OK, response = ApiResponseMessage.class),
+			@ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST)
+	})
+	public ResponseEntity<ApiResponseMessage> deleteAuthor(@PathVariable("authorId") Integer authorId,
+			@AuthenticationPrincipal UserPrincipal userPrincipal) {
+		ApiResponseMessage apiResponseMessage;
+		apiResponseMessage = ApiResponseMessage.builder()
+				.success(authorService.delete(authorId, userPrincipal))
+				.message(String.format("Doctor %s deleted successfully", authorId))
+				.build();
+		return ResponseEntity.ok().body(apiResponseMessage);
+	}
+
+	@GetMapping(DOCTORS)
```

Comment:
Let's make it similar to PostController. The name should be `DOCTOR_ALL_DOCTORS` and the value `/all-doctors`
![image](https://user-images.githubusercontent.com/86650723/185445996-889f6890-d0ed-460d-95c9-03d36f63935f.png)

Reply:
fixed